pulp_rpm/app/advisory.py

```diff
@@ -190,6 +192,25 @@ def resolve_advisory_conflict(previous_advisory, added_advisory):
     return to_add, to_remove, to_exclude
 
 
+def _copy_update_collections_for(the_collections):
+    """
+    Deep-copy each UpdateCollection in the_collections, and its UpdateCollectionPackages.
+    """
+    new_collections = []
+    for collection in the_collections:
+        uc_packages = collection.packages.all()
+        collection.pulp_id = None
+        collection.pk = None
+        collection.save()
+        for a_package in uc_packages:
+            a_package.pulp_id = None
+            a_package.pk = None
+            a_package.update_collection = collection
+            a_package.save()
+        new_collections.append(collection)
```

Comment:
Any chance you use a bulk update instead of ```.save()``` both here and above?

Reply:
the collection has to be saved before you can "point to it" from the new package instance. I can probably bulk-save the packages tho, will experiment.

Reply:
Consider doing all of this in a transaction as well, couldn't hurt and still reduces the # of round-trips.

Reply:
Put _copy_update_collections work inside a transaction